applications/asset_tracker_v2/src/cloud/cloud_codec/aws_iot_codec.c

```diff
@@ -20,6 +20,35 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(cloud_codec, CONFIG_CLOUD_CODEC_LOG_LEVEL);
 
+/* Function that checks the version number of the incoming message and determines if it has already
+ * been handled. Receiving duplicate messages can often occur upon retransmissions from the AWS IoT
+ * broker due to unACKed MQTT QoS 1 messages and unACKed TCP packets. Messages from AWS IoT
+ * shadow topics contains an incrementing version number.
+ */
+static bool has_message_been_handled(cJSON *root_obj)
```

Comment:
```suggestion
static bool has_shadow_update_been_handled(cJSON *root_obj)
```

What I don't like here is that the API has not relation to the shadow topic, and `version_prev` is used globally.
I think the global aspect is fine (after all there will be only one shadow per device), but I'd like to make sure that it's clear that this API is intended for use with the shadow.

Reply:
Agreed, specifying this with the function name makes this much clearer.